packages/block-editor/src/components/block-list/block.js

```diff
@@ -636,14 +646,18 @@ const applyWithSelect = withSelect(
 			getBlockOrder,
 			__unstableGetBlockWithoutInnerBlocks,
 			isNavigationMode,
+			getBlockHierarchyRootClientId,
 		} = select( 'core/block-editor' );
+
 		const block = __unstableGetBlockWithoutInnerBlocks( clientId );
+
 		const isSelected = isBlockSelected( clientId );
 		const { hasFixedToolbar, focusMode, isRTL } = getSettings();
 		const templateLock = getTemplateLock( rootClientId );
 		const isParentOfSelectedBlock = hasSelectedInnerBlock( clientId, true );
 		const index = getBlockIndex( clientId, rootClientId );
 		const blockOrder = getBlockOrder( rootClientId );
+		const isRootOfHierarchy = clientId === getBlockHierarchyRootClientId( clientId );
```

Comment:
I think we can not rely on isRootOfHierarchy and need a more complex condition.
If the navigation block is nested several levels deep e.g: inside the columns. This logic will make the toolbar of the navigation menu item appear on the columns block while the toolbar of the navigation menu itself appears bellow on the block.
The toolbar should not appear on the top-level block but on the closes ancestor that used consumeChildToolbar.

Reply:
You make a good point. I need to start thinking about Columns more and nesting.

> ...the closes ancestor that used consumeChildToolbar.

What tools are available to us to access the settings for each `InnerBlocks`? 

Also this assumes that we're going to have `consumeChildToolbar ` apply recursively. In hindsight, I'm thinking perhaps that would be best but I'm not 100% sure. There might be hidden use cases I'm overlooking where you'd need fine grain control.